runtime/common/src/crowdloan/migration.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_support::generate_storage_alias;
+
+/// Migrations for using fund index to create fund accounts instead of para ID.
+pub mod crowdloan_index_migration {
+	use super::*;
+
+	// The old way we generated fund accounts.
+	fn old_fund_account_id<T: Config>(index: ParaId) -> T::AccountId {
+		T::PalletId::get().into_sub_account(index)
+	}
+
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+		// `NextTrieIndex` should have a value.
+		generate_storage_alias!(Crowdloan, NextTrieIndex => Value<FundIndex>);
+		let next_index = NextTrieIndex::get().unwrap_or_default();
+		ensure!(next_index > 0, "Next index is zero, which implies no migration is needed.");
+
+		log::info!(
+			target: "runtime",
+			"next trie index: {:?}",
+			next_index,
+		);
```

Comment:
As per offline discussion assuming this is a debug artifact that will get removed before merging

Reply:
Well I guess actually doesn't really matter since this is feature gated to try-runtime builds

Reply:
yeah, i actually liked having it, and the log is only in the `pre` and `post` check